src/webgpu/shader/execution/expression/call/builtin/normalize.spec.ts

```diff
@@ -10,47 +10,31 @@ import { makeTestGroup } from '../../../../../../common/framework/test_group.js'
 import { GPUTest } from '../../../../../gpu_test.js';
 import { TypeF32, TypeF16, TypeVec } from '../../../../../util/conversion.js';
 import { FP } from '../../../../../util/floating_point.js';
-import { vectorF32Range, vectorF16Range } from '../../../../../util/math.js';
 import { makeCaseCache } from '../../case_cache.js';
 import { allInputSources, run } from '../../expression.js';
 
 import { builtin } from './builtin.js';
 
 export const g = makeTestGroup(GPUTest);
 
-// Cases: f32_vecN_[non_]const
-const f32_vec_cases = ([2, 3, 4] as const)
-  .flatMap(n =>
-    ([true, false] as const).map(nonConst => ({
-      [`f32_vec${n}_${nonConst ? 'non_const' : 'const'}`]: () => {
-        return FP.f32.generateVectorToVectorCases(
-          vectorF32Range(n),
-          nonConst ? 'unfiltered' : 'finite',
-          FP.f32.normalizeInterval
-        );
-      },
-    }))
-  )
-  .reduce((a, b) => ({ ...a, ...b }), {});
-
-// Cases: f16_vecN_[non_]const
-const f16_vec_cases = ([2, 3, 4] as const)
-  .flatMap(n =>
-    ([true, false] as const).map(nonConst => ({
-      [`f16_vec${n}_${nonConst ? 'non_const' : 'const'}`]: () => {
-        return FP.f16.generateVectorToVectorCases(
-          vectorF16Range(n),
-          nonConst ? 'unfiltered' : 'finite',
-          FP.f16.normalizeInterval
-        );
-      },
-    }))
+const vec_cases = (['f32', 'f16'] as const)
+  .flatMap(trait =>
+    ([2, 3, 4] as const).flatMap(dim =>
+      ([true, false] as const).map(nonConst => ({
+        [`${trait}_vec${dim}_${nonConst ? 'non_const' : 'const'}`]: () => {
+          return FP[trait].generateVectorToVectorCases(
+            FP[trait].vectorRange(dim),
+            nonConst ? 'unfiltered' : 'finite',
+            FP[trait].normalizeInterval
+          );
+        },
+      }))
+    )
   )
   .reduce((a, b) => ({ ...a, ...b }), {});
 
 export const d = makeCaseCache('normalize', {
```

Comment:
Just 
``` suggestion
export const d = makeCaseCache('normalize', vec_cases);
```
is enough. Same for other cases.

Reply:
Done